woocommerce/woocommerce-reset
# Initial Commit

Adds a plugin providing a REST API endpoint to reset WooCommerce data.

For now, this deletes WP Options used by WC Admin.

The planned use-case for the endpoint is to provide isolation for the WooCommerce E2E tests, by enabling data to be reset between tests.

### Testing Instructions

- Install this as a plugin. Download the zip from http://github.com/woocommerce/woocommerce-reset/zipball/init/
- Make a HTTP request to `POST /woocommerce-reset-tool/v1/reset`. For example:
```
curl -X "POST" "http://localhost:8000/index.php?rest_route=/woocommerce-reset/v1/reset"
```
- See that options such as `woocommerce_admin_install_timestamp` and `woocommerce_allow_tracking` do not exist.

#### Unit Tests

I've added PHP unit tests, which leverages wp-env to provide the WordPress testing environment. This is similar to Gutenberg’s approach. To run them:

```
composer install
npm install
npm run test
```
